runtime/jvmti/jvmtiObject.c

```diff
@@ -206,59 +209,60 @@ jvmtiGetObjectMonitorUsage(jvmtiEnv* env,
 }
 
-
 static void
-findMonitorThreads(J9VMThread * vmThread, J9JVMTIMonitorStats * pStats)
+findMonitorThreads(J9VMThread *vmThread, J9JVMTIMonitorStats *pStats)
 {
-	j9object_t lockObject;
+	j9object_t lockObject = NULL;
+	J9VMThread *currentThread = (J9VMThread *)pStats->currentThread;
+	j9object_t target = (j9object_t)vmThread->threadObject;
+	UDATA threadState = 0;
+
+	if (NULL == target) {
+		return;
+	}
+
+#if JAVA_SPEC_VERSION >= 19
+	if (IS_JAVA_LANG_VIRTUALTHREAD(currentThread, target)) {
+		return;
+	}
+#endif /* JAVA_SPEC_VERSION >= 19 */
 
-	UDATA threadState = getVMThreadObjectStatesAll(vmThread, &lockObject, NULL, NULL);
+	threadState = getVMThreadObjectStatesAll(vmThread, &lockObject, NULL, NULL);
 
 	/* Stopped on the same monitor object? */
 	if (lockObject == pStats->lockObject) {
-
-		j9object_t target = (j9object_t)vmThread->threadObject;
-
-		if (target) {
-
-			J9JavaVM * vm = pStats->vm;
-			JNIEnv * jniEnv = (JNIEnv *)pStats->currentThread;
-
-			/* CMVC 87023 - the spec is unclear, but from experimentation it appears that 'waiting to be notified' threads
-			 * should appear in both lists
-			 */
-			threadState &= ~(J9VMTHREAD_STATE_SUSPENDED | J9VMTHREAD_STATE_INTERRUPTED);
-			switch (threadState) {
-				case J9VMTHREAD_STATE_WAITING:
-				case J9VMTHREAD_STATE_WAITING_TIMED:
-					if (pStats->waiting == NULL) {
-						(pStats->numWaiting)++;
-					} else {
-						if (pStats->waitingCntr < pStats->numWaiting) {
-							pStats->waiting[pStats->waitingCntr] =
-									(jthread) vm->internalVMFunctions->j9jni_createLocalRef(jniEnv, target);
-							++(pStats->waitingCntr);
-						}
+		J9JavaVM *vm = pStats->vm;
+		JNIEnv *jniEnv = (JNIEnv *)pStats->currentThread;
+
+		threadState &= ~(J9VMTHREAD_STATE_SUSPENDED | J9VMTHREAD_STATE_INTERRUPTED);
+		switch (threadState) {
+			case J9VMTHREAD_STATE_WAITING:
+			case J9VMTHREAD_STATE_WAITING_TIMED:
+				if (pStats->waiting == NULL) {
+					(pStats->numWaiting)++;
+				} else {
+					if (pStats->waitingCntr < pStats->numWaiting) {
+						pStats->waiting[pStats->waitingCntr] =
+								(jthread) vm->internalVMFunctions->j9jni_createLocalRef(jniEnv, target);
+						++(pStats->waitingCntr);
 					}
-					/* FALL THROUGH */
-
-				case J9VMTHREAD_STATE_BLOCKED:
-					if (pStats->blocked == NULL) {
-						(pStats->numBlocked)++;
-					} else {
-						if (pStats->blockedCntr < pStats->numBlocked) {
-							pStats->blocked[pStats->blockedCntr] =
-									(jthread) vm->internalVMFunctions->j9jni_createLocalRef(jniEnv, target);
-							++(pStats->blockedCntr);
-						}
+				}
+				break;
+
+			case J9VMTHREAD_STATE_BLOCKED:
+				if (pStats->blocked == NULL) {
+					(pStats->numBlocked)++;
+				} else {
+					if (pStats->blockedCntr < pStats->numBlocked) {
+						pStats->blocked[pStats->blockedCntr] =
+								(jthread) vm->internalVMFunctions->j9jni_createLocalRef(jniEnv, target);
+						++(pStats->blockedCntr);
```

Comment:
Should we change to `(pStats->blockedCntr) += 1;`?

Reply:
Updated.